exec/java-exec/src/main/java/org/apache/drill/exec/record/AbstractRecordBatch.java

```diff
@@ -228,4 +228,20 @@ public WritableBatch getWritableBatch() {
   public VectorContainer getOutgoingContainer() {
     throw new UnsupportedOperationException(String.format(" You should not call getOutgoingContainer() for class %s", this.getClass().getCanonicalName()));
   }
+
+  public void drainStream(IterOutcome stream, int input, RecordBatch batch) {
+    if (stream == IterOutcome.OK_NEW_SCHEMA || stream == IterOutcome.OK) {
+      for (final VectorWrapper<?> wrapper : batch) {
+        wrapper.getValueVector().clear();
+      }
+      batch.kill(true);
+      stream = next(input, batch);
+      while (stream == IterOutcome.OK_NEW_SCHEMA || stream == IterOutcome.OK) {
+        for (final VectorWrapper<?> wrapper : batch) {
+          wrapper.getValueVector().clear();
+        }
+        stream = next(input, batch);
+      }
```

Comment:
In one-on-one discussions it became clear that fixing this issue is out of the scope of this particular task. That's fine. Please just file a JIRA to track the potential problem so we don't forget about it.